homeassistant/components/openalpr.py

```diff
@@ -0,0 +1,395 @@
+            except ImportError:
+                _LOGGER.error("Local openalpr instalation not exists")
+                continue
+
+            alpr_dev = OpenalprDeviceLocal(
+                name=device.get(CONF_NAME),
+                render=render,
+                interval=device.get(CONF_INTERVAL),
+                input_source=input_source,
+                runtime=device.get(CONF_RUNTIME),
+                region=region,
+                extra_arguments=device.get(CONF_EXTRA_ARGUMENTS),
+                username=device.get(CONF_USERNAME),
+                password=device.get(CONF_PASSWORD),
+            )
+
+        else:
+            alpr_dev = OpenalprDeviceCloud(
+                name=device.get(CONF_NAME),
+                render=render,
+                interval=device.get(CONF_INTERVAL),
+                input_source=input_source,
+                api_key=device.get(CONF_RUNTIME),
+                region=region,
+                extra_arguments=device.get(CONF_EXTRA_ARGUMENTS),
+                username=device.get(CONF_USERNAME),
+                password=device.get(CONF_PASSWORD),
+            )
+
+        # register shutdown event
+        openalpr_device.append(alpr_dev)
+        hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, alpr_dev.shutdown)
+
+    component.add_entities(openalpr_device)
+
+    def _handle_service_scan(service):
+        """Handle service for immediately scan."""
+        device_list = component.extract_from_service(service)
+
+        for device in device_list:
+            device.scan()
+
+    hass.services.register(DOMAIN, SERVICE_SCAN,
+                           _handle_service_scan,
+                           descriptions[DOMAIN][SERVICE_SCAN],
+                           schema=SERVICE_SCAN_SCHEMA)
+
+    # Add restart service only if a device use ffmpeg as render
+    if not use_render_fffmpeg:
+        return True
+
+    def _handle_service_restart(service):
+        """Handle service for restart ffmpeg process."""
+        device_list = component.extract_from_service(service)
+
+        for device in device_list:
+            device.restart()
+
+    hass.services.register(DOMAIN, SERVICE_RESTART,
+                           _handle_service_restart,
+                           descriptions[DOMAIN][SERVICE_RESTART],
+                           schema=SERVICE_RESTART_SCHEMA)
+
+    return True
+
+
+# pylint: disable=too-many-instance-attributes
+class OpenalprDevice(Entity):
+    """Represent a openalpr device object for processing stream/images."""
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, name, render, interval, input_source,
+                 extra_arguments=None, username=None, password=None):
+        """Init image processing."""
+        from haffmpeg import ImageStream
+
+        self._name = name
+        self._render = render
+        self._interval = interval
+        self._last = set()
+
+        # generade input source
+        if self._render == RENDER_FFMPEG:
```

Comment:
I've move the api check to voluptous. For the render spliting I need to look at code how I want do that.